### PR TITLE
Added Christmas Balls

### PR DESCRIPTION
**Pull request by gregalexan**
Christmas is over however it's not too late. I added Christmas Balls to the holiday card!

### CODE REVIEW

File: src/main/java/gr/aueb/card/Tree.java [hunk 1]
```diff
@@ -104,20 +106,40 @@ private void drawTrunk(Graphics g) {
...
-	g.fillPolygon(p);
-	g.setColor(c); // Revert paint color to default
+	    /* Fill it with green color */
+	    Color c = g.getColor();
+	    g.setColor(green);
+	    g.fillPolygon(p);
+	    g.setColor(c); // Revert paint color to default
+
+        /* Adding Christmas Balls to the body of the tree */
+        drawBalls(dimensions.x -2, (int) (dimensions.y + dimensions.height * bodyHeightFactor) -2,
```
Comment (dspinellis) [inline src/main/java/gr/aueb/card/Tree.java]: There is a lot of boilerplate code in the calls below. Can you express it more succinctly by moving functionality to the `drawBall` (better name) method?

File: src/main/java/gr/aueb/card/Tree.java [hunk 2]
```diff
@@ -45,6 +45,8 @@ public class Tree extends Drawable {
     private final Color brown = new Color(204, 102, 0);
     /** Body's color (RGB) */
     private final Color green = new Color(0, 254, 0);
+    /** Christmas Balls' color (RGB) */
```
Comment (dspinellis) [inline src/main/java/gr/aueb/card/Tree.java]: Tree balls'

File: src/main/java/gr/aueb/card/Tree.java [hunk 3]
```diff
@@ -104,20 +106,40 @@ private void drawTrunk(Graphics g) {
...
+        //   10, 10, g); // Middle down
+        drawBalls(dimensions.x + dimensions.width / 2 -5 - 20, (int) (dimensions.y + dimensions.height * bodyHeightFactor) -5 - 20,
+                10, 10, g); //Up Left
+        drawBalls(dimensions.x + dimensions.width / 2 -5 +20, (int) (dimensions.y + dimensions.height * bodyHeightFactor) -5 -20,
+                10, 10, g); //Up Right
+        drawBalls(dimensions.x + dimensions.width / 2 -5, (int) (dimensions.y + dimensions.height * bodyHeightFactor) -5 - 60,
+                10, 10, g); //Middle
+
+    }
+    private void drawBalls(int x, int y, int width, int height, Graphics g) {
```
Comment (dspinellis) [inline src/main/java/gr/aueb/card/Tree.java]: * [ ] Please add JavaDoc comment
* [ ] Write comments in imperative voice
* [ ] Move g argument to be the first

**Review by dspinellis — CHANGES_REQUESTED** [review]
Nice, well done! I'd like to integrate your change. Please
* [ ] Consider the comments I made
* [ ] Remove the .idea files (they are IDE-specific, hence not relevant)
* [ ] Squash your commits into one again

**Comment by gregalexan** [comment]
Added one more commit, shall i open a new pull request?

**Comment by dspinellis** [comment]
> Added one more commit, shall i open a new pull request?

You need to squash the two (`git rebase -i`) and then force-push.

**Comment by gregalexan** [comment]
Hope nothing went wrong

File: src/main/java/gr/aueb/card/Tree.java [hunk 4]
```diff
@@ -104,20 +106,61 @@ private void drawTrunk(Graphics g) {
...
+
+    /**
+     * Calls addTreeOrnament for specific locations on
+     * the tree body.
+     * @param g The Graphics object on which we will paint
+     */
+    private void setTreeOrnaments(Graphics g) {
+        /* Add ornament to down left. */
+        addTreeOrnament(g, dimensions.x -2,
+                (int) (dimensions.y + dimensions.height * bodyHeightFactor) -2);
```
Comment (dspinellis) [inline src/main/java/gr/aueb/card/Tree.java]: Consider pushing the repeated code `  (int) (dimensions.y + dimensions.height * bodyHeightFactor)` to the called function.

File: src/main/java/gr/aueb/card/Tree.java [hunk 5]
```diff
@@ -104,20 +106,61 @@ private void drawTrunk(Graphics g) {
...
+        /* Fill Tree Ornament with color. */
+        g.fillOval(x, y, 10, 10);
+    }
+
+    /**
+     * Calls addTreeOrnament for specific locations on
+     * the tree body.
+     * @param g The Graphics object on which we will paint
+     */
+    private void setTreeOrnaments(Graphics g) {
```
Comment (dspinellis) [inline src/main/java/gr/aueb/card/Tree.java]: `set` is used for setters. `draw` is a better prefix.

**Review by dspinellis — CHANGES_REQUESTED** [review]
I added two comments. Also please [squash all your commits into one](https://www.git-tower.com/learn/git/faq/git-squash).

BTW this is only needed if you want me to integrate your changes. You already have the exercise bonus.

**Pull request merged**